### PR TITLE
Issue #3032407 by ribel: Refactor page hero block

### PR DESCRIPTION
**Pull request by ribel**
## Problem
Currently, it's confusing and hard to extend or override `page_hero_data` template if you want to have a different hero for different content types.

## Solution
Move hero logic from `page-hero-data.html.twig` to `node--hero.html.twig` and use `hero_node` variable in `page_hero_data` template.

## Issue tracker
- https://www.drupal.org/project/social/issues/3032407
- https://jira.goalgorilla.com/browse/GPEDC-72

## How to test
- [ ] Checkout to this branch, revert features and clear the cache.
- [ ] Check that node hero looks the same as before
- [ ] Copy `node--hero.html.twig` to `node--event--hero.html.twig`, and add some text or made some changes there
- [ ] Clear the cache and open Event
- [ ] You should see your changes in the hero.
- [ ] Hero for Topics should stay the same

## Release notes
Before, it was hard for developers to have a different hero for different content types. We moved hero logic from `page-hero-data.html.twig` to `node--hero.html.twig` so now it's easier to extend it.

### CODE REVIEW

**Comment by robertragas** [comment]
I checked code-wise and looks good, need to functional test it later when solution for event managers is in.

**Comment by Kingdutch** [comment]
The target branch of this PR has been changed to 6.x which will be the next major version. Our policy is to merge any bugfixes in the newest branch first and cherry-pick/backport any required changes. This ensures that the newest major version is the longest/least diverged branch.

**Comment by jaapjan** [comment]
@ribel considering all the failing tests and the fact that it is quite a big change I propose to keep this for a later 5.x release. What do you think?

**Comment by ribel** [comment]
@jaapjan Yes, I agree. 
We will try to fix tests soon, and after that, I would like to get a review from the distro team.

**Comment by jaapjan** [comment]
Marking as `needs work`

File: modules/social_features/social_core/social_core.module [hunk 1]
```diff
@@ -36,27 +36,15 @@ function social_core_theme() {
...
- *   - hero_node: Rendered hero display of the current active node or NULL.
+ * Implements hook_preprocess_HOOK().
  */
-function template_preprocess_page_hero_data(array &$variables) {
-  // Get current user.
-  $account = \Drupal::currentUser();
-
-  // Get current node object.
+function social_core_preprocess_node(&$variables) {
+  /** @var \Drupal\node\Entity\Node $node */
```
[the remark this answers is not in the document]
Comment (ribel) [reply modules/social_features/social_core/social_core.module]: Will be done

File: modules/social_features/social_core/src/Plugin/Block/SocialPageTitleBlock.php [hunk 2]
```diff
@@ -148,6 +148,7 @@ public function build() {
           '#theme' => 'page_hero_data',
           '#title' => $title,
           '#node' => $node,
+          '#hero_node' => node_view($node, 'hero'),
```
[the remark this answers is not in the document]
Comment (ribel) [reply modules/social_features/social_core/src/Plugin/Block/SocialPageTitleBlock.php]: Will be done.

**Comment by ribel** [comment]
I have also implemented improvements proposed by @OH17L.

File: modules/social_features/social_core/src/Plugin/Block/SocialPageTitleBlock.php [hunk 3]
```diff
@@ -70,13 +78,16 @@ class SocialPageTitleBlock extends PageTitleBlock implements ContainerFactoryPlu
    *   The entity repository.
    * @param \Drupal\Core\Controller\TitleResolverInterface $title_resolver
    *   The title resolver.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entityTypeManager
```
Comment (OH17L) [inline modules/social_features/social_core/src/Plugin/Block/SocialPageTitleBlock.php]:
```suggestion
   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
```

File: modules/social_features/social_core/src/Plugin/Block/SocialPageTitleBlock.php [hunk 4]
```diff
@@ -70,13 +78,16 @@ class SocialPageTitleBlock extends PageTitleBlock implements ContainerFactoryPlu
...
+   *   The entity type manager.
    */
-  public function __construct(array $configuration, $plugin_id, $plugin_definition, RouteMatchInterface $route_match, RequestStack $request_stack, EntityRepositoryInterface $entity_repository, TitleResolverInterface $title_resolver) {
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, RouteMatchInterface $route_match, RequestStack $request_stack, EntityRepositoryInterface $entity_repository, TitleResolverInterface $title_resolver, EntityTypeManagerInterface $entityTypeManager) {
```
Comment (OH17L) [inline modules/social_features/social_core/src/Plugin/Block/SocialPageTitleBlock.php]:
```suggestion
  public function __construct(array $configuration, $plugin_id, $plugin_definition, RouteMatchInterface $route_match, RequestStack $request_stack, EntityRepositoryInterface $entity_repository, TitleResolverInterface $title_resolver, EntityTypeManagerInterface $entity_type_manager) {
```

File: modules/social_features/social_core/src/Plugin/Block/SocialPageTitleBlock.php [hunk 5]
```diff
@@ -70,13 +78,16 @@ class SocialPageTitleBlock extends PageTitleBlock implements ContainerFactoryPlu
...
     parent::__construct($configuration, $plugin_id, $plugin_definition);
     $this->routeMatch = $route_match;
     $this->requestStack = $request_stack;
     $this->entityRepository = $entity_repository;
     $this->titleResolver = $title_resolver;
+    $this->entityTypeManager = $entityTypeManager;
```
Comment (OH17L) [inline modules/social_features/social_core/src/Plugin/Block/SocialPageTitleBlock.php]:
```suggestion
    $this->entityTypeManager = $entity_type_manager;
```

File: themes/socialbase/templates/node/page-hero-data.html.twig [hunk 6]
```diff
@@ -1,65 +1,18 @@
-{#
+{#	{{ hero_node }}
```
[the remark this answers is not in the document]
Comment (ribel) [reply themes/socialbase/templates/node/page-hero-data.html.twig]: It is already there

[the remark this answers is not in the document]
Comment (OH17L) [reply themes/socialbase/templates/node/page-hero-data.html.twig]: Then just delete it here. Is it needed here?

**Comment by ribel** [comment]
@ronaldtebrake @jaapjan This one is ready for your final review and merge.

**Comment by ronaldtebrake** [comment]
Hi guys, works like a charm so I will merge it. 
Could you please also add a change record on drupal.org explaining to our distro users what it means for their subthemes. 

Thanks in advance :) !

For example what we did in 5.0: 
https://www.drupal.org/node/3019929

**Pull request merged**